### PR TITLE
small changes on linear regression

### PR DESCRIPTION
**Pull request by Quid37**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Improvement

Short description (up to few sentences):
Functions `linearRegression`, `logisticRegression` return model weights when not used as State in function `evalMLMethod`. Added performance test for `linearRegression`. Added documentation for `linearRegression`.

### CODE REVIEW

File: dbms/src/AggregateFunctions/AggregateFunctionMLMethod.cpp [hunk 1]
```diff
@@ -149,6 +149,16 @@ void LinearModelData::predict(
     gradient_computer->predict(container, block, arguments, weights, bias, context);
 }
 
+void LinearModelData::return_weights(IColumn & to) const
```
Comment (alexey-milovidov) [inline dbms/src/AggregateFunctions/AggregateFunctionMLMethod.cpp]: Style.

File: dbms/tests/performance/linear_regression.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    <create_query>CREATE TABLE train_dataset(p1 Float64, p2 Float64, target Float64) ENGINE = Memory</create_query>
+    <create_query>CREATE TABLE test_dataset(p1 Float64, p2 Float64) ENGINE = Memory</create_query>
+
+    <create_query>CREATE TABLE test_model engine = Memory as select linearRegressionState(0.001)(target, p1, p2) as state from train_dataset</create_query>
+
+    <fill_query>INSERT INTO train_dataset values (2.0, 1.0, 6.0), (3.0, -1.0, 6.0), (4.0, 2.0, 11.0), (-2.0, 1.0, -2.0)</fill_query>
+    <fill_query>INSERT INTO test_dataset values (1.0, 1.0), (5.0, -3.0)</fill_query>
+
+<!--    Проверяем, как работает fit-->
```
Comment (alexey-milovidov) [inline dbms/tests/performance/linear_regression.xml]: Comments must be only in english (both in code and in tests).

File: dbms/tests/performance/linear_regression.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    <main_metric>
+        <min_time/>
+    </main_metric>
+
+    <create_query>CREATE TABLE train_dataset(p1 Float64, p2 Float64, target Float64) ENGINE = Memory</create_query>
+    <create_query>CREATE TABLE test_dataset(p1 Float64, p2 Float64) ENGINE = Memory</create_query>
+
+    <create_query>CREATE TABLE test_model engine = Memory as select linearRegressionState(0.001)(target, p1, p2) as state from train_dataset</create_query>
+
+    <fill_query>INSERT INTO train_dataset values (2.0, 1.0, 6.0), (3.0, -1.0, 6.0), (4.0, 2.0, 11.0), (-2.0, 1.0, -2.0)</fill_query>
```
Comment (alexey-milovidov) [inline dbms/tests/performance/linear_regression.xml]: Table is way too small. Better to reuse Taxi, Ontime or Metrica dataset
(look at the used tables across other performance tests).

There are many useful queries like "predict ride price according to time, distance and number of passengers".

File: dbms/src/AggregateFunctions/AggregateFunctionMLMethod.cpp [hunk 4]
```diff
@@ -149,6 +149,16 @@ void LinearModelData::predict(
...
+{
+    auto & column = static_cast<ColumnVector<Float64> &>(to);
+    for (auto weight_value : weights)
+    {
+        column.getData().push_back(weight_value);
```
Comment (alexey-milovidov) [inline dbms/src/AggregateFunctions/AggregateFunctionMLMethod.cpp]: And this is totally incorrect. You're returning Float64 not Array(Float64).

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

File: dbms/src/AggregateFunctions/AggregateFunctionMLMethod.cpp [hunk 5]
```diff
@@ -149,6 +151,29 @@ void LinearModelData::predict(
...
+
+    size_t old_size = offsets_to.back();
+    offsets_to.push_back(old_size + size);
+
+    typename ColumnFloat64::Container & val_to
+            = static_cast<ColumnFloat64 &>(arr_to.getData()).getData();
+
+    val_to.reserve(old_size + size);
+    size_t i = 0;
+    while (i < weights.size())
```
Comment (KochetovNicolai) [inline dbms/src/AggregateFunctions/AggregateFunctionMLMethod.cpp]: Use for loop.

File: dbms/src/AggregateFunctions/AggregateFunctionMLMethod.h [hunk 6]
```diff
@@ -269,7 +272,15 @@ class AggregateFunctionMLMethod final : public IAggregateFunctionDataHelper<Data
     {
     }
 
-    DataTypePtr getReturnType() const override { return std::make_shared<DataTypeNumber<Float64>>(); }
+    DataTypePtr getReturnType() const override
+    {
+        return std::make_shared<DataTypeArray>(std::make_shared<DataTypeFloat64>());
+    }
+
+    DataTypePtr getReturnTypeToPredict() const override
```
Comment (KochetovNicolai) [inline dbms/src/AggregateFunctions/AggregateFunctionMLMethod.h]: Comment is needed

**Pull request merged**